Adds a new pixel to mitigate crash bursts over-reporting

diegoreymendez opened the pull request:
Task/Issue URL: https://app.asana.com/1/137249556945/project/1201037661562251/task/1211381214122815?focus=true

### Description

Adds a new pixel to report crash events rather than single tab crashes.  The issue with looking at single tab crashes is we get excess tab crash reports simply because the user has more tabs.

### Testing Steps

1. Enable FF: Debug Menu > Feature Flag Overrides > Other > tabCrashDebugging
2. Open Console.app and filter by "category:PixelKit any:webkit"
3. Open a few tabs and load sites on them
4. Go to Debug Menu > Simulate Crash > Crash All Tabs (if this is not present you might need to restart the app - it's controlled by the feature flag above)
5. The tabs should crash.
6. Crash the tabs again within 5 seconds - this is necessary because that makes the crash non-recoverable.
You should see something like this in Console.app, where the aggregated pixel shows only once and reports the correct tab count:
```
👾[Daily and Count-Fired] m_mac_webkit_did_terminate_aggregated_daily ["appVersion": "1.159.0", "crash_count": "3", "pixelSource": "browser-dmg"]
👾[Daily and Count-Fired] m_mac_webkit_did_terminate_aggregated ["appVersion": "1.159.0", "crash_count": "3", "pixelSource": "browser-dmg"]
```

### Impact and Risks

Low.  This is a new pixel so not affecting anything that already exists.

#### What could go wrong?

Unclear... I'd think the biggest risk might be in the aggregator not properly handling the debouncing.

### Quality Considerations

NA

### Notes to Reviewer

NA

---
###### Internal references:
[Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f) | [Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552) | [Tech Design Template](https://app.asana.com/0/59792373528535/184709971311943)


<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Introduce burst tab-crash aggregation with a new pixel and plumbing through Tab/TabExtensions, plus a debug action to crash all tabs.
> 
> - **Telemetry**:
>   - Add `TabCrashAggregator` to detect and debounce burst WebKit tab crashes (100ms) and fire aggregated pixel `webkit_did_terminate_non_recoverable_aggregated` with `tab_count`.
>   - Extend `GeneralPixel` with `webKitDidTerminateNonRecoverableAggregated` and add pixel definition in `macOS/PixelDefinitions/pixels/webkit_termination_pixels.json5`.
> - **Tab crash handling**:
>   - Inject `TabCrashAggregator` via `AppDelegate`, `Tab` initializers, and `TabExtensionDependencies`; wire into `TabCrashRecoveryExtension` to record crashes on loops/debug cases.
> - **Debug menu**:
>   - Add "Crash All Tabs" action (behind `.tabCrashDebugging`) to kill all tab web content processes.
> - **Project/Tests**:
>   - Add `TabCrashAggregator.swift` to project and sources; update unit tests to pass aggregator.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 6c2686d9ba5c84f1950cd57fb73215aa1ffbaba7. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

jozsef-vesza commented on macOS/DuckDuckGo/Tab/TabExtensions/TabCrashAggregator.swift:
```diff
@@ -0,0 +1,65 @@
+
+        debounceTask?.cancel()
+        debounceTask = Task.detached(priority: .utility) {
+            try? await Task.sleep(nanoseconds: 100 * NSEC_PER_MSEC)
```
![](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fwww.looper.com%2Fimg%2Fgallery%2Fthe-history-of-spider-mans-pointing-meme-explained%2Fl-intro-1686144123.jpg&f=1&nofb=1&ipt=bc8f333ecfd2f18b78fcb6a8aec8a8aef091f3f5633b8e70fa09e1ed43402056)

diegoreymendez replied:
Resolved in latest commit.

diegoreymendez commented on macOS/DuckDuckGo/Tab/TabExtensions/TabCrashAggregator.swift:
```diff
@@ -0,0 +1,65 @@
+            await MainActor.run {
+                self.fireAggregatedPixel()
+            }
+        }
```
Resolved.

diegoreymendez commented on macOS/DuckDuckGo/Tab/TabExtensions/TabCrashAggregator.swift:
```diff
@@ -0,0 +1,70 @@
+        debounceTask?.cancel()
+        debounceTask = Task.detached(priority: .utility) {
+            do {
+                try await Task.sleep(nanoseconds: 100 * NSEC_PER_MSEC)
```
Now cancels correctly.

diegoreymendez replied:
Will change this so it can't extend forever.

diegoreymendez commented:
**Changes:**
- Addressed your feedback.
- Changed the trigger for this - we want this to fire for non-recoverable tab crashes.
- Resolved an issue with a pre-existing pixel where it was using an unsupported trigger in its definition (CI was complaining about it so I was forced to update it).

SabrinaTardio commented on macOS/DuckDuckGo/Statistics/GeneralPixel.swift:
```diff
@@ -1094,6 +1095,9 @@ enum GeneralPixel: PixelKitEvent {
         case .webKitTerminationIndicatorClicked:
             return "webkit_termination_indicator_clicked"
+        /// Event trigger: Aggregated WebKit process crashes (burst detection)
+        case .webKitDidTerminateAggregated:
```
Maybe we can specify it is a non recoverable termination?
Just in case we want to measure others in the fure
